src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -30,177 +47,513 @@ function NameTokenSelector({
+            pdnsSourceContract.records,
+          );
+          return {
+            ...result,
+            [id.toString()]: {
+              owner,
+              controller,
+              name,
+              ticker,
+              names: associatedNames,
+            },
+          };
+        },
+        {},
+      );
+      setTokens(newTokens);
+      if (imports) {
+        const details = newTokens[imports[0].toString()];
+        setSelectedToken({
+          name: details?.name,
+          ticker: details?.ticker,
+          id: imports[0].toString(),
+        });
+      }
+    } catch (error: any) {
+      eventEmitter.emit('error', error);
+    } finally {
+      setLoading(false);
+      setListPage(1);
+    }
+  }
+
+  function handleTokenSearch(query: string) {
     try {
       setSearching(true);
-      if (!id.length) {
+      setValidImport(undefined);
+      if (!query) {
         setSearchText('');
+        setSelectedToken(undefined);
         return;
       }
+      setSearchText(query);
 
-      setSearchText(id);
-      const txid = new ArweaveTransactionID(id);
-      selectedTokenCallback(txid);
-      const state =
-        await arweaveDataProvider.getContractState<PDNTContractJSON>(txid);
-      const contract = new PDNTContract(state, txid);
-
-      if (!contract.isValid()) {
-        throw Error('Contract does not match required schema');
+      if (!tokens) {
+        throw new Error('No Name Tokens Found');
       }
-      setSelectedNameToken({
-        id,
-        name: contract.name,
-        ticker: contract.ticker,
-      });
-      setValidImport(true);
-      setSearchText('');
+      const filteredResults = Object.keys(tokens)
+        .filter((id) => {
+          const { owner, controller, name, ticker } = tokens[id];
+          const queryResult = [owner, controller, name, ticker, id].some(
+            (term) =>
+              term &&
+              term.toString().toLowerCase().includes(query.toLowerCase()),
+          );
+          return queryResult;
+        })
+        .map((id) => {
+          const { name, ticker } = tokens[id];
+          return { id, name: name ?? '', ticker: ticker ?? '' };
+        })
+        .filter((n) => n !== undefined);
+      if (!filteredResults.length) {
+        throw new Error('No ANT tokens found for that search');
+      }
+
+      setFilteredTokens(filteredResults);
     } catch (error) {
+      setFilteredTokens(undefined);
       eventEmitter.emit('error', error);
-      setSelectedNameToken(undefined);
-      setValidImport(false);
     } finally {
       setSearching(false);
+      setListPage(1);
     }
   }
 
+  function handleSetToken({
+    id,
+    name,
+    ticker,
+  }: {
+    id?: string;
+    name?: string;
+    ticker?: string;
+  }) {
+    setSearchText('');
+    setFilteredTokens(undefined);
+
+    if (id === undefined) {
+      if (selectedToken === undefined) {
+        setSearchActive(false);
+      }
+      return;
+    }
```

Comment:
i'm a bit confused by this logic - if `id` is `undefined` but `selectedToken` is not we return? can we consolidate to something like:
```suggestion
if (!id && !selectedToken) {
   setSearchActive(false);
   return;
}
```

Reply:
it is required to verify id is there before proceeding which needs to be an independent check.

I moved this to just the id check, wrapped in try-catch-finally